### ISSUE TITLE
The useObject doesn't re-render when the value transitions from undefined to an object.

### ISSUE THREAD

**Issue by decisionnguyen** [realm/realm-js#6101]
### How frequently does the bug occur?

Always

### Description
Hi Team,

I'm using Redux to cache an array of IDs. However, when the Realm database is cleared, the re-rendering doesn't occur when the IDs are updated. I have conducted debugging, and I've observed that...

The originalObjectRef.current in the useObject sometimes returns undefined, not null. However, within the function, it's checked using a null value.

In line **81** always return undefined
But in line **85** it's **"originalObjectRef.current === null"**, when I change to **"!originalObjectRef.current"** => it's work fine

![image](https://github.com/realm/realm-js/assets/22584353/1b808df9-fa6c-40fe-9a02-041888eebfa2)

```
diff --git a/node_modules/@realm/react/src/useObject.tsx b/node_modules/@realm/react/src/useObject.tsx
index 3d438b1..6d5d189 100644
--- a/node_modules/@realm/react/src/useObject.tsx
+++ b/node_modules/@realm/react/src/useObject.tsx
@@ -81,7 +81,7 @@ export function createUseObject(useRealm: () => Realm) {
         // Re-instantiate the cachedObject if the primaryKey has changed or the originalObject has gone from null to not null
         if (
           !arePrimaryKeysIdentical(primaryKey, primaryKeyRef.current) ||
-          (originalObjectRef.current === null && originalObject !== null)
+          (!originalObjectRef.current && originalObject !== null)
         ) {
           cachedObjectRef.current = createCachedObject({
             object: originalObject ?? null,

```

### Can you reproduce the bug?

Always

### Reproduction Steps

"realm": "11.0.0-rc.0",
"@realm/react": "^0.5.2",
"react-native": "0.68.5",

### Version

11.0.0-rc.0

### What services are you using?

Local Database only

### Are you using encryption?

Yes

### Platform OS and version(s)

0.68.5

### Build environment

Which debugger for React Native: ..

**Comment by takameyer** [comment]
@decisionnguyen We have recently released some fixes in `@realm/react` surrounding `useObject`.  Can you update to `0.6.0` and verify that the issue still persists?

**Comment by takameyer** [comment]
@decisionnguyen You will need to upgrade `realm`.  There were API changes made to the stable `11.0.0` version that aren't included in the very alpha Release Candidate your are currently using (`11.0.0-rc.0`).  I actually tried testing against this version and many of my tests failed. One of the changes included is that `Realm.objectForPrimaryKey` returns `null` instead of `undefined`.
Another important change is we are using a different value to cache realm objects, which wasn't available in version `11.0.0-rc.0`.  This will cause a crash in your application with:
```
object._objectKey is not a function
```

I

**Comment by decisionnguyen** [comment]
Thank you very much
Wish you all the best
